code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -1466,6 +1468,14 @@ static HookFunction initFunction([]()
 		GetTrainDoor(train, doorIndex)->ratio = ratio;
 	}));
 
+	fx::ScriptEngine::RegisterNativeHandler("SET_TRAIN_STOP_AT_STATIONS", makeTrainFunction([](fx::ScriptContext& context, fwEntity* train)
+	{
+		bool state = context.GetArgument<bool>(1);
+		*(int*)((uintptr_t)train + g_trainFlagOffset) |= 16 * (state & 1);
```

Comment:
This won't work to set the state to false.

You can use something similar to how we handle `vehicleTankEmptyFlags`:
```
		uint8_t* trainFlags = (uint8_t*)((uintptr_t)train + g_trainFlagOffset);
		*trainFlags ^= (*trainFlags ^ (uint8_t)state) & 1;
```

Reply:
I actually found that we have a function to set a vehicle bit. So we can replace this native implementation with:
```
fx::ScriptEngine::RegisterNativeHandler("SET_TRAIN_STOP_AT_STATIONS", std::bind(writeVehicleMemoryBit<&g_trainFlagOffset, 4>, _1, "SET_TRAIN_STOP_AT_STATIONS"));
```

It looks better that way.